src/components/pages/courseDetailPage.tsx

```diff
@@ -1,7 +1,146 @@
+
+    const [loading, setLoading] = useState(true);
+    const [course, setCourse] = useState({} as Course);
+    const [userCourse, setUserCourse] = useState<UserCourse>({} as UserCourse);
+    const [assignments, setAssignments] = useState(new Array<Assignment>());
+    const [error, setError] = useState(null);
+    const [orderBy, setOrderBy] = useState<OrderBy>(defaultOrderBy);
+    const [groupBy, setGroupBy] = useState<GroupBy>(defaultGroupBy);
+
+    const { courseId } = useParams() as Params;
+
+    useEffect(() => {
+        fetchCourse(orderBy, groupBy);
+    }, []);
+
+    const userId = useAppSelector((store) => store.user.id);
+
+    const fetchCourse = async (orderBy: OrderBy, groupBy: GroupBy) => {
+        try {
+            const pageCourse = await RequestService.get(`/api/courses/${courseId}`);
+            setCourse(pageCourse);
+            setAssignments(await RequestService.get(`/api/assignments?orderBy=${orderBy}?groupBy=${groupBy}`));
+
+            // Find the users relationship to the course by calling all user-course objects, and seeing if any have the user/course id.
+            const userCourses = await RequestService.get<UserCourse[]>(`/api/user-courses`);
+            const uCourse = userCourses.filter(u => u.courseId === parseInt(courseId) && u.userId === userId).shift()
+            if (uCourse) { setUserCourse(uCourse); }
+        }
+        catch(error) {
+            setError(error)
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    const handleFilterChange = (updatedOrderBy: OrderBy) => {
+        setOrderBy(updatedOrderBy)
+        fetchCourse(updatedOrderBy, groupBy)
+    
+        LocalStorageService.set(ORDER_BY_STORAGE_KEY, updatedOrderBy)
+      }
+    
+      const handleGroupByChange = (updatedGroupBy: GroupBy) => {
+        setGroupBy(updatedGroupBy)
+        fetchCourse(orderBy, updatedGroupBy)
+    
+        LocalStorageService.set(GROUP_BY_STORAGE_KEY, updatedGroupBy)
+      }
+
+    if (loading) return <LoadingOverlay delay={250} />
+    if (error) return <ErrorPage error={error} />
+
+    const defaultOrderByOption = orderByOptions.find((o) => o.value === orderBy)
+    const defaultGroupByOption = groupByOptions.find((o) => o.value === groupBy)
+
+    return (
+        <PageWrapper>
+            <header>
+                <h1>{course.name}</h1>
+                <div className={styles.subtitle}>
+                    <h3>{course.number} | {course.semester} ({prettyPrintDate(course.startDate)}-{prettyPrintDate(course.endDate)})</h3>
+                    {userCourse.level === "instructor" &&
+                        <Link to={`${courseId}/update`}><Button>Update Course</Button></Link>
+                    } 
+                </div>
+            </header>
+            {userCourse &&
+                <div>
+                    <Link className={styles.button} to={`${courseId}/assignments`}><Button>Gradebook</Button></Link>
```

Comment:
I'm not entirely sure this is the correct link, but I was having issues finding a grade book page